[Cppcheck] The document for Sider's recommended set of rules and add suppressions-list option (#632)
sider/sider-docs: docs/tools/cplusplus/cppcheck.md

$ cppcheck --help
```

## Default Configuration for Cppcheck

Sider provides our [recommended ruleset](https://github.com/sider/runners/blob/HEAD/images/cppcheck/sider_recommended_cppcheck.txt) for Cppcheck.

> This configuration is used when you do not have `sider.yml` in your repository or the option `suppressions-list` in the configuration.

Reviewer: ```suggestion
This configuration is used when the `suppressions-list` option is not specified in `sider.yml`.
```

This statement implies the default configuration will be enabled if sider.yml is not there.

## Default Configuration for Cppcheck

Sider provides our [recommended ruleset](https://github.com/sider/runners/blob/HEAD/images/cppcheck/sider_recommended_cppcheck.txt) for Cppcheck.

> This configuration is used This configuration is used when the `suppressions-list` option is not specified in `sider.yml`.

Reviewer: ```suggestion
This configuration is used when the [`suppressions-list`](#suppressions-list) option is not specified in `sider.yml`.
```
- remove duplicate words "This configuration is used"
- Link to the `suppressions-list` section on this page

Author: Thanks, I fixed it.

- unusedFunction
- Whole program analysis (ctu=Cross Translation Unit)

### `suppressions-list`

> This option allows you to set the file path of ignoring for rules.
> In the file, you describe one rule per line like below.
> 
> ```
> nullPointer
> ctunullpointer
> ```

Reviewer: How about this? This file format is so flexible that it would be good to tell users a link to the reference manual.

Also, note that `null` in `nullPointer` is highlighted on the generated document for some reason. Take another examples such as memleak and uninitvar. Or, "```text" may work.

```suggestion
This option allows you to specify a file path of a suppressions file. You can create a suppressions file for example as follows:

memleak
uninitvar

You can read more about this file format in the [Cppcheck manual](https://github.com/danmar/cppcheck/blob/main/man/manual.md#suppressions-in-a-file).
```

Author: Thanks, I fixed it.

### `suppressions-list`

This option allows you to specify a file path of a suppressions file. You can create a suppressions file for example as follows:

```text
memleak
uninitvar
```

> You can read more about this file format in the [Cppcheck manual](https://github.com/danmar/cppcheck/blob/main/man/manual.md#suppressions-in-a-file

Reviewer: ```suggestion
You can read more about this file format in the [Cppcheck manual](https://github.com/danmar/cppcheck/blob/main/man/manual.md#suppressions-in-a-file).
```

missing the closing parenthesis and period.

Author: Thanks, I fixed it.
